pkg/controlplane/authz/manager.go

```diff
@@ -218,20 +228,36 @@ func (m *Manager) getPodInfoByIP(ip string) *podInfo {
 	return nil
 }
 
+func (m *Manager) getClientAttributes(req *egressAuthorizationRequest) connectivitypdp.WorkloadAttrs {
+	clientAttrs := connectivitypdp.WorkloadAttrs{GatewayNameLabel: m.getPeerName()}
+	podInfo := m.getPodInfoByIP(req.IP)
+	if podInfo == nil {
+		m.logger.Infof("Pod has no info: IP=%v.", req.IP)
+		return clientAttrs // better return an error here?
```

Comment:
Returning an error will simply yield a RST to the client connection.
I think it's better to keep it the way it is (which still allows for allow-all policies to proceed).

Reply:
This means we are not zero-trust by default.
To implement zero-trust, users will have to add a deny policy, blocking connections from unknown pods.
E.g.:
```yaml
apiVersion: clusterlink.net/v1alpha1
kind: PrivilegedAccessPolicy
metadata:
  name: deny-unknown-pods
spec:
  action: deny
  from:
  - workloadSelector:
      matchExpressions:
      - key: clusterlink/metadata.clientNamespace
        operator: Exists 
  to:
  - workloadSelector: {}
```

Reply:
This is a bit problematic when having policies with negative selectors (using the `NotIn` and `DoesNotExist` operators).
E.g., I have a policy to deny all connections, except from those originating from namespace `foo`.
Connections from IPs for which CL cannot link a Pod, will always be allowed.
```yaml
apiVersion: clusterlink.net/v1alpha1
kind: AccessPolicy
metadata:
  name: deny-except-from-foo
spec:
  action: deny
  from:
  - workloadSelector:
      matchExpressions:
      - key: clusterlink/metadata.clientNamespace
        operator: NotIn
        value: foo
  to:
  - workloadSelector: {}
```

Reply:
Right.
I think the right thing to do is to do a non-cached GET from the API-server to get the pod info in this case.
I did not find a way to do this via the given client. This will be the best option.

Reply:
Found this:
https://github.com/kubernetes-sigs/controller-runtime/issues/585#issuecomment-528102351

So there's a non-cached reader, but it's not available to `authz.Manager`. 
Need to propagate it from `cl-controlplane/app/server.go`.

Reply:
Apparently, even this is not good enough. Pod may start running and send requests, even before its IP is updated in etcd. From what I see, some non-cached `List()` calls return the Pod with its Node IP, but without its Pod IP. It only gets updated later.

Reply:
How long is this window of a working pod with etcd not yet updated?
If not long, maybe we can stall a bit to wait for the update.

Reply:
I changed the implementation: now if the client has no attributes (no pod info) AND the PDP has at least one policy which depends on client attributes, the request will be denied, and the client will have to retry.
Note that even if the PDP has no attribute-dependent policies, attribute-less requests can still be denied, depending on the policies (e.g., DENY policies take precedence, or no ALLOW policies)
This is enforced both on egress and on ingress.